services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/model/ReadBatch.java

```diff
@@ -27,6 +27,14 @@
 import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
 import software.amazon.awssdk.services.dynamodb.model.KeysAndAttributes;
 
+/**
+ * Defines a collection of primary keys for items in a table, stored as {@link KeysAndAttributes}.
+ * <p/>
+ *
+ * <b>Note:</b>The builder for the class is parameterized, however the class itself is not. The builder takes a parameterized
+ * {@link MappedTableResource} in order for the constructor to transform {@link GetItemEnhancedRequest} objects into the keys and
+ * attributes, but the object itself only stores a reference to the table name and does not need to remember the type.
```

Comment:
I think we should cut this, we don't need to explain our implementation in public javadoc if it's not relevant to usability.

Reply:
Cut.